app/src/components/mainWindow.ts

```diff
@@ -155,7 +155,11 @@ export async function createMainWindow(
   }
 
   if (options.targetUrl) {
-    await mainWindow.loadURL(options.targetUrl);
+    try {
+      await mainWindow.loadURL(options.targetUrl);
+    } catch (err: unknown) {
+      log.error('targetUrl ERROR', err);
+    }
   }
```

Comment:
Hi, thanks for the PR 🙂. In #1415 you write

> _"This was due to undefined `mainWindow` variable."_

Sooo, why go for a try/catch? What about

```ts
if (options.targetUrl && mainWindow)
```

Reply:
Sorry for not being specific. The one I'm mentioning was this variable is getting set as `undefined`.

https://github.com/nativefier/nativefier/blob/master/app/src/main.ts#L331

The function is returning undefined as `loadURL` was getting error.

Reply:
@spywhere 👍, thanks for the precision, and thanks for your research in #1415. This try/catch still feels a bit fishy to me. I don't have a Mac to play with it and come up with better things to try.

@TheCleric could you help here?

Reply:
@TheCleric ping about that; would love your halp with that since I don't have a Mac.

Reply:
So looking at this, there does not seem to be a good way to do this HERE without the try/catch.

However I found a slightly "cleaner" way of doing it.

If we just delete the loadURL call out of `createMainWindow` and place it at the end of `onReady()` like so:

`return mainWindow.loadURL(appArgs.targetUrl);`

That way `createMainWindow`always returns a handle, and then we can let `onReady` (and it's callers) worry about what they'd like to do about handling the error (which right now is simply logging it).

I'm not COMPLETELY happy with this, but it does remove the try/catch plus allows the exposure of the error in case we need to do something with it later.

If any of this is unclear @ronjouch or @spywhere please let me know, and I can submit a PR with what I was thinking.

Reply:
That's definitely better than try/catch. I'm good with this 👍

Reply:
> That's definitely better than try/catch. I'm good with this +1

@spywhere great! Go ahead with the implementation; will merge.

Thx @TheCleric for the idea.

Reply:
PR updated

Reply:
> PR updated

@spywhere you sure? I see you did move `loadURL` to the end of `onReady`, but @TheCleric was suggesting *returning* the result of `loadURL`.

Reply:
I was consider doing that as well. Though the return type of `createMainWindow` is `Promise<void>` and returning the value from `loadURL` is not change anything, I thought it might give a misunderstanding that `createMainWindow` might requires `loadURL` to run last. Hence using `await` instead to allow more changes to be added after the `loadURL` call.

Reply:
> I was consider doing that as well. Though the return type of `createMainWindow` is `Promise<void>` and returning the value from `loadURL` is not change anything, I thought it might give a misunderstanding that `createMainWindow` might requires `loadURL` to run last. Hence using `await` instead to allow more changes to be added after the `loadURL` call.

Alright, fine by me. Merging, thanks!